name.abuchen.portfolio/src/name/abuchen/portfolio/snapshot/security/DividendCalculation.java

```diff
@@ -10,77 +15,185 @@
+            this.amount = t.getGrossValue().with(converter.at(t.getDateTime()));
+            LocalDateTime time = t.getDateTime();
+            this.year = time.getYear();
+            this.date = time.toLocalDate();
+        }
+
+        @Override
+        public int compareTo(DividendPayment o)
+        {
+            return this.date.compareTo(o.date);
+        }
+    }
 
     private LocalDate firstPayment;
     private LocalDate lastPayment;
-
-    private int regularEvents;
-
-    @Override
-    public void setTermCurrency(String termCurrency)
-    {
-        super.setTermCurrency(termCurrency);
-        this.sum = MutableMoney.of(termCurrency);
-    }
+    private final List<DividendPayment> payments = new ArrayList<DividendPayment>();
+    private Periodicity periodicity;
+    private MutableMoney sum;
 
     @Override
-    public void visit(CurrencyConverter converter, DividendTransaction t)
+    public void finish()
     {
-        sum.add(t.getGrossValue().with(converter.at(t.getDateTime())));
-        numOfEvents++;
-
-        if (t.getShares() > 0
-                        && (lastPayment == null || Dates.daysBetween(lastPayment, t.getDateTime().toLocalDate()) > 30))
+        // first sort
+        Collections.sort(payments);
+        // default is unknown periodicity
+        this.periodicity = Periodicity.UNKNOWN;
+        // check payments?
+        if (!payments.isEmpty())
         {
-            regularEvents++;
+            // get first and last payment
+            firstPayment = payments.get(0).date;
+            lastPayment = payments.get(payments.size() - 1).date;
+
+            HashMap<Integer, MutableMoney> hmSumPerYear = new HashMap<Integer, MutableMoney>();
+            // sum up dividends
+            for (DividendPayment p : payments)
+            {
+                // add total sum
+                sum.add(p.amount);
+                // add sum per year
+                MutableMoney m = hmSumPerYear.get(p.year);
+                if (m == null)
+                {
+                    // construct if missing
+                    m = MutableMoney.of(sum.getCurrencyCode());
+                    hmSumPerYear.put(p.year, m);
+                }
+                m.add(p.amount);
+            }
+
+            int significantCount = 0;
+            // now walk through individual years
+            for (int year = firstPayment.getYear(); year <= lastPayment.getYear(); year++)
+            {
+                int count = 0;
+                // first calc sum for year
+                MutableMoney m = MutableMoney.of(sum.getCurrencyCode());
+                for (DividendPayment p : payments)
+                {
+                    if (p.year == year)
+                    {
+                        m.add(p.amount);
+                        count++;
+                    }
+                }
+                double sum = m.getAmount();
+                // expected amount
+                double expectedAmount = sum / count;
```

Comment:
Ich glaube ```count``` kann hier tatsächlich 0 sein wenn es es Jahr gibt in dem es tatsächlich keine Dividende gab. Also zB. wenn die mal komplett ausgesetzt wurde.